### ISSUE TITLE
Negative `ListView.builder(itemCount: ..)` breaks childrenInHitTestOrder assertion

### ISSUE THREAD

**Issue by anfelor** [flutter/flutter#45500]
## Steps to Reproduce

1. Use a negative number as the `itemCount`.
2. Example application on https://gist.github.com/anfelor/e1e60cd227b5d4f8ac06fa0ec5879fbe
  Second button triggers the bug below.

**Target Platform:** Android
**Target OS version/browser:** Android 10
**Devices:** Only tested on physical

## Logs

```
[+22146 ms] I/flutter (17486): ══╡ EXCEPTION CAUGHT BY SCHEDULER LIBRARY ╞═════════════════════════════════════════════════════════
[        ] I/flutter (17486): The following assertion was thrown during a scheduler callback:
[        ] I/flutter (17486): If a node has scrollChildren, it must have childrenInHitTestOrder
[        ] I/flutter (17486): 'dart:ui/semantics.dart':
[        ] I/flutter (17486): Failed assertion: line 713: '<optimized out>'
[        ] I/flutter (17486): 
[        ] I/flutter (17486): Either the assertion indicates an error in the framework itself, or we should provide substantially
[        ] I/flutter (17486): more information in this error message to help you determine and fix the underlying cause.
[        ] I/flutter (17486): In either case, please report this assertion by filing a bug on GitHub:
[        ] I/flutter (17486):   https://github.com/flutter/flutter/issues/new?template=BUG.md
[        ] I/flutter (17486): 
[        ] I/flutter (17486): When the exception was thrown, this was the stack:
[        ] I/flutter (17486): #2      SemanticsUpdateBuilder.updateNode (dart:ui/semantics.dart:713:7)
[        ] I/flutter (17486): #3      SemanticsNode._addToUpdate (package:flutter/src/semantics/semantics.dart:2066:13)
[        ] I/flutter (17486): #4      SemanticsOwner.sendSemanticsUpdate (package:flutter/src/semantics/semantics.dart:2650:14)
[        ] I/flutter (17486): #5      PipelineOwner.flushSemantics (package:flutter/src/rendering/object.dart:1042:23)
[        ] I/flutter (17486): #6      RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:348:19)
[        ] I/flutter (17486): #7      WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:774:13)
[        ] I/flutter (17486): #8      RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:283:5)
[        ] I/flutter (17486): #9      SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1097:15)
[        ] I/flutter (17486): #10     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1036:9)
[        ] I/flutter (17486): #11     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:952:5)
[        ] I/flutter (17486): #15     _invoke (dart:ui/hooks.dart:259:10)
[        ] I/flutter (17486): #16     _drawFrame (dart:ui/hooks.dart:217:3)
[        ] I/flutter (17486): (elided 5 frames from class _AssertionError and package dart:async)
[        ] I/flutter (17486): ════════════════════════════════════════════════════════════════════════════════════════════════════
```

```
Analyzing flutter_bug...                                                
No issues found! (ran in 1.8s)
```

```
[✓] Flutter (Channel master, v1.12.8-pre.17, on Linux, locale de_DE.utf8)
    • Flutter version 1.12.8-pre.17 at /home/anton/.local/bin/flutter
    • Framework revision 4c64b715d9 (3 days ago), 2019-11-21 22:12:24 -0800
    • Engine revision 7e68097cc1
    • Dart version 2.7.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /home/anton/Android/Sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /home/anton/.local/bin/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[!] Android Studio (version 3.5)
    • Android Studio at /home/anton/.local/bin/android-studio
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.40.1)
    • VS Code at /usr/share/code
    • Flutter extension version 3.6.0

[✓] Connected device (3 available)
    • HD1903     • 7d6b23ca   • android-arm64  • Android 10 (API 29)
    • Chrome     • chrome     • web-javascript • Chromium 78.0.3904.97 Built on Ubuntu , running on Ubuntu 19.04
    • Web Server • web-server • web-javascript • Flutter Tools

! Doctor found issues in 1 category.
```

**Comment by dnfield** [comment]
To be clear - the fix for this is asserting earlier.  Having a negative item count is invalid.  You should update your code to make sure you never give a negative item count (perhaps with something like `math.max(0, possiblyNegativeNumber)`.

**Comment by anfelor** [comment]
@dnfield Sure :) Props for fixing this so fast!
